geo-types/src/lib.rs

```diff
@@ -64,61 +64,72 @@ extern crate rstar_0_8;
 #[macro_use]
 extern crate approx;
 
-#[deprecated(since = "0.7.0", note = "use `CoordFloat` or `CoordNum` instead")]
-pub trait CoordinateType: Num + Copy + NumCast + PartialOrd + Debug {}
-#[allow(deprecated)]
-impl<T: Num + Copy + NumCast + PartialOrd + Debug> CoordinateType for T {}
-
 /// The type of an x or y value of a point/coordinate.
 ///
 /// Floats (`f32` and `f64`) and Integers (`u8`, `i32` etc.) implement this.
 ///
 /// For algorithms which only make sense for floating point, like area or length calculations,
 /// see [CoordFloat](trait.CoordFloat.html).
-#[allow(deprecated)]
-pub trait CoordNum: CoordinateType + Debug {}
-#[allow(deprecated)]
-impl<T: CoordinateType + Debug> CoordNum for T {}
+pub trait CoordNum: Num + Copy + NumCast + PartialOrd + Debug {}
+impl<T: Num + Copy + NumCast + PartialOrd + Debug> CoordNum for T {}
 
 pub trait CoordFloat: CoordNum + Float {}
 impl<T: CoordNum + Float> CoordFloat for T {}
 
+pub trait ZCoord: NumOps + Zero + Copy + PartialEq + PartialOrd + Debug {}
```

Comment:
We should add docs for the `ZCoord` and `Measure` traits

Also, how come the trait bounds here (`NumOps + Zero + ...`) differ from `CoordNum`?

Reply:
> Also, how come the trait bounds here (`NumOps` + `Zero` + ...) differ from `CoordNum`?

Actually, is there any reason why we couldn't make `ZCoord` just extend `CoordNum`?

Reply:
In order to support `CoordNum` in `NoValue`, we would need to implement `ToPrimitive`, `NumCast`, and `Num` traits. Possible implementation below.  I am worried though that it will cause more harm than good -- i.e. `Num::from_str_radix()` shouldn't allow all possible values, etc. Also, having fewer traits ensures that algorithms that don't work with NoValues are not accidentally used in that way - e.g. someone wouldn't be able to create a `CoordTZM<NoValue, f64, NoValue>` (only Z value is given) -- we expect coordinate to always have real x,y values.

```rust
impl ToPrimitive for NoValue {
    fn to_i64(&self) -> Option<i64> {
        Some(0)
    }

    fn to_u64(&self) -> Option<u64> {
        Some(0)
    }
}

impl NumCast for NoValue {
    fn from<T: ToPrimitive>(_: T) -> Option<Self> {
        Some(Self::default())
    }
}

impl Num for NoValue {
    type FromStrRadixErr = ();

    fn from_str_radix(str: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
        Ok(Self::default())
    }
}
```

Reply:
Thanks for the response, I think I understand. I am going to check out your git branch in the next day or two and experiment with this so I have a better grasp of this